src/patternfly/components/FileUpload/examples/FileUpload.md

```diff
@@ -0,0 +1,150 @@
+      {{> file-upload-text-input
+        file-upload-text-input--aria-label="Filename"
+        file-upload-text-input--attribute='readonly value="Sample.txt"'
+        }}
+      {{#> button button--modifier="pf-m-control"}}
+        Browse...
+      {{/button}}
+      {{#> button button--modifier="pf-m-control"}}
+        Clear
+      {{/button}}
+    {{/input-group}}
+  {{/file-upload-input-group}}
+  {{#> file-upload-textarea file-upload-textarea--aria-label="Text area"}}Ssh-Rsa AAh3zJFkzjjakCJialksjfB3zJFkzzAAhhMskjjakCJialksjfB3z89z3zJFkz3 +kzMAjsauoox88aaZXphBx4fczJFkzMAjsauoox88aaZXphBx4fczJFkzMAjsauoox88aaZXphBx4fc
+  {{/file-upload-textarea}}
+{{/file-upload}}
+```
+
+```hbs title=Drag-file-hover-component
+{{#> file-upload file-upload--id="drag-file-hover-component" file-upload--modifier="pf-m-drag-hover"}}
+  {{#> file-upload-input-group}}
+    {{#> input-group}}
+      {{> file-upload-text-input file-upload-text-input--aria-label="Drag or drop a file" file-upload-text-input--attribute='placeholder="Drag a file here or browse to upload"'}}
+      {{#> button button--modifier="pf-m-control"}}
+        Browse...
+      {{/button}}
+      {{#> button button--modifier="pf-m-control" button--attribute="disabled"}}
+        Clear
+      {{/button}}
+    {{/input-group}}
+  {{/file-upload-input-group}}
+  {{#> file-upload-textarea file-upload-textarea--aria-label="Empty text area"}}
+  {{/file-upload-textarea}}
+{{/file-upload}}
+```
+
+```hbs title=File-upload-error
+{{#> file-upload file-upload--id="file-upload-error"}}
+  {{#> file-upload-input-group}}
+    {{#> input-group}}
+      {{> file-upload-text-input
+        file-upload-text-input--aria-label="File upload error"
+        file-upload-text-input--attribute='required value="Sample.png" aria-invalid="true"'
+        }}
+      {{#> button button--modifier="pf-m-control"}}
+        Browse...
+      {{/button}}
+      {{#> button button--modifier="pf-m-control"}}
+        Clear
+      {{/button}}
+    {{/input-group}}
+  {{/file-upload-input-group}}
+  {{#> file-upload-textarea file-upload-textarea--aria-label="Empty text area"}}{{/file-upload-textarea}}
+  {{#> file-upload-message file-upload-message--modifier="pf-m-error"}}
+    {{#> form-helper-text form-helper-text--modifier="pf-m-error" form-helper-text--attribute='id="file-upload-error-helper-text" aria-live="polite"'}}
+      We don't support this file type. Try again with a different file type.
+    {{/form-helper-text}}
+  {{/file-upload-message}}
+{{/file-upload}}
+```
+
+```hbs title=File-upload-loading
+{{#> file-upload file-upload--id="file-upload-loading" file-upload--modifier="pf-m-loading"}}
+  {{#> file-upload-input-group}}
+    {{#> input-group}}
+      {{> file-upload-text-input
+        file-upload-text-input--aria-label="Read only"
+        file-upload-text-input--attribute='readonly name="file-upload-loading" value="Sample.png"'
+      }}
+      {{#> button button--modifier="pf-m-control"}}
```

Comment:
Yep treating it so that when the file is loading, you can still browse for a file which will override the current file, or you can cancel.

Reply:
I think this would be handled by the React implementation but I would not expect the user to be able to click Browse while an upload is in progress. I just seems like it over complicates the React code. I'd say to disable Browse in this state. Do you agree @mturley ?

Reply:
ok, maybe for the react implementation we should throw a warning that if they browse they will cancel their current upload. It could be confused that you can upload more than one file rather than replace the current upload.

Reply:
I don't have a strong opinion on this, but it makes sense to me to block the Browse button during an upload since we have the Clear button which could function as a cancel action.

While the React component should support the asynchronous upload case for those who need it, I expect the majority use case (directly loading a small text file from the user's local machine into a form) to effectively happen instantly, and the in-progress view would never appear. In that case, the actual transfer of the file contents to a server would happen when the form is submitted, not when the file is selected.

In fact, I think that might be reason enough to call this component "File Input" rather than "File Upload".

Reply:
Based on everyone's thoughts I think it would be best to disable the browse button while the upload is happening.

@mcarrano what are your thoughts on @mturley name suggestion?